dspback/triggers.py

```diff
@@ -77,6 +77,8 @@ async def watch_submissions():
                     result = await db["discovery"].delete_one({"repository_identifier": document["identifier"]})
                     logger.warning(f"delete count {result.deleted_count}")
             else:
+                # TODO: This is not going to work probably, as we don't have a document object in the case of a
```

Comment:
Wondering how we can delete a record from discovery when we find a deleted submission.

Reply:
Ah, nice catch.  That should be `change["documentKey"]["_id"]`.  Would you go ahead and update it? We must have not been testing this by deleting submissions, rather by toggling private/public or deleting the record at the repository directly.

Reply:
So this needs to be changed to:
`await db["discovery"].delete_one({"repository_identifier": change["documentKey"]["_id"]]})`

I assume `change["documentKey"]["_id"]` gives us the id of the deleted submission. Then we can't use that id to delete a record in discovery right?

Reply:
you're right.  Thanks for thinking that through before making the change.  I was looking at the watch_discovery function above `await db["typeahead"].delete_one({"_id": change["documentKey"]["_id"]})`.  But now you have me second guessing that one too.  It is doing an _id match from discovery to typeahead.  That works because tyepahead is using the _id from discovery on insert to typeahead.

I think for this line,we can probably just move the `public_json_ld = retrieve_submission_json_ld(document)` to before the if statement.

Reply:
I'll fire it up and run some tests.

Reply:
https://github.com/cznethub/dspback/pull/179